src/pytorch_lightning/core/saving.py

```diff
@@ -229,7 +237,9 @@ def _load_state(
 
         # 2. Try to restore model hparams from checkpoint using the new key
         _new_hparam_key = cls.CHECKPOINT_HYPER_PARAMS_KEY
-        cls_kwargs_loaded.update(checkpoint.get(_new_hparam_key))
```

Comment:
Would this work?

```py
cls_kwargs_loaded.update(checkpoint.get(_new_hparam_key, {}))
```

Reply:
It should, but mypy complained. I can comment with the error this resolved in the morning. I don't exactly remember but I'm pretty sure it was an arg-type error.

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13932/commits/99643a0ed8ad27dd5bc94a5eaf7070523223b5dd

Reply:
What I meant was, to leave the code as it is in master but add the default value {} in the get() call.  

```py
_new_hparam_key = cls.CHECKPOINT_HYPER_PARAMS_KEY
cls_kwargs_loaded.update(checkpoint.get(_new_hparam_key), {})  # <--- here see added {}
```

This doesn't require any None checks, thus much cleaner IMO.

Reply:
updated in https://github.com/Lightning-AI/lightning/pull/13932/commits/9735d21c777fa8fa70c0d89f7c971940baff4fa2